### ISSUE TITLE
Error 500 : An error occurred while saving the entity changes. See the inner exception for details

### ISSUE THREAD

**Issue by antoine-dhersin** [Azure/Commercial-Marketplace-SaaS-Accelerator#739]
**Describe the bug**
When we try to subscribe to an offer, we receive the following error: "Error 500 : An error occurred while saving the entity changes. See the inner exception for details". We user the URL & the token received but unable to finalize the subscription.

**To Reproduce**
Try to subscribe to : https://appsource.microsoft.com/en-us/product/web-apps/openlaketechnology1695221806668.ms_adoption_monitor?tab=Overview

**Expected behavior**
Being able to subscribe after purchase.

**Screenshots**
![Capture d’écran 2024-07-28 234643](https://github.com/user-attachments/assets/9823f38f-4609-4d85-b197-997f1195ff0c)

**Environment (please complete the following information):**
Latest code update, deployed with cloud shell
    
**Additional context**

**Comment by antoine-dhersin** [comment]
Hi, anyone who can help? I've updated the app to the latest version, removed any special characters in the offer... Other offers are ok for the subscription, I just have an issue with one offer...

**Comment by santhoshb-msft** [comment]
Hi @antoine-dhersin I believe this could be still due to the length or special chars, Is there any way you can do a local debug and see what the issue is?

**Comment by antoine-dhersin** [comment]
Could you please tell me how to activate the local debug?

**Comment by santhoshb-msft** [comment]
Hi @antoine-dhersin - can you see if this can help - https://partner.microsoft.com/en-us/training/assets/detail/setting-up-a-development-environment-for-the-saas-accelerator-mp4

**Comment by antoine-dhersin** [comment]
Hi,

I'm not sure it may help. My customer was able to subscribe to 4 offers without any issue, but one offer is not working.

Is there a way to set a debug on the app to trace what's happening?

Thanks, Antoine.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by santhoshb-msft** [comment]
Please follow the video and I recommend creating a ticket on partner center.

**Comment by antoine-dhersin** [comment]
I just went through the video and can't find any option to enable the logging or something else... Is there a way to read the logs inside the app? If yes, what's the path?

**Comment by santhoshb-msft** [comment]
@antoine-dhersin please keep us posted on how the preview purchase went

**Comment by antoine-dhersin** [comment]
Hi,

Not working, we even try to publish the offer... The end customer is still having a 500 error....

Thanks, Antoine.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by santhoshb-msft** [comment]
where you able to do a preview test?

**Comment by antoine-dhersin** [comment]
Yes, it was not working but we went to live offer anyway to try...

Can we review the offer again? (I cross checked offer name and specific characters...)

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ssquadri** [comment]
You have to delete the complete Resource group and its deployed resources, Then Go to Key Vault > Manage deleted keys > Select subscription >Purge that key.

After that remove the directory of commercial marketplace in Terminal by rm -rf command and then finally re-run the script.

We were able to solve the Error 500 issue of landing page by following these steps, It might work for you as well : )
